MediaBrowser.Common/Net/IPHost.cs

```diff
@@ -0,0 +1,444 @@
+                    {
+                        output += HostName + ",";
+                    }
+                    else if (i.Equals(IPAddress.Any))
+                    {
+                        output += "Any IP4 Address,";
+                    }
+                    else if (Address.Equals(IPAddress.IPv6Any))
+                    {
+                        output += "Any IP6 Address,";
+                    }
+                    else if (i.Equals(IPAddress.Broadcast))
+                    {
+                        output += "Any Address,";
+                    }
+                    else
+                    {
+                        output += $"{i}/32,";
+                    }
+                }
+
+                output = output[0..^1];
+
+                if (moreThanOne)
+                {
+                    output += "]";
+                }
+            }
+            else
+            {
+                output = HostName;
+            }
+
+            return output;
+        }
+
+        /// <inheritdoc/>
+        public override void Remove(AddressFamily family)
+        {
+            if (ResolveHost())
+            {
+                _addresses = _addresses.Where(p => p.AddressFamily != family).ToArray();
+            }
+        }
+
+        /// <inheritdoc/>
+        public override bool Contains(IPObject address)
+        {
+            // An IPHost cannot contain another IPObject, it can only be equal.
+            return Equals(address);
+        }
+
+        /// <inheritdoc/>
+        protected override IPObject CalculateNetworkAddress()
+        {
+            var netAddr = NetworkAddressOf(this[0], PrefixLength);
+            return new IPNetAddress(netAddr.Address, netAddr.PrefixLength);
+        }
+
+        /// <summary>
+        /// Attempt to resolve the ip address of a host.
+        /// </summary>
+        /// <returns><c>true</c> if any addresses have been resolved, otherwise <c>false</c>.</returns>
+        private bool ResolveHost()
+        {
+            // When was the last time we resolved?
+            if (_lastResolved == 0)
+            {
+                _lastResolved = DateTime.UtcNow.Ticks;
+            }
+
+            // If we haven't resolved before, or out timer has run out...
+            if ((_addresses.Length == 0 && !Resolved) || (TimeSpan.FromTicks(DateTime.UtcNow.Ticks - _lastResolved).TotalMinutes > Timeout))
+            {
+                _lastResolved = DateTime.UtcNow.Ticks;
+                ResolveHostInternal().GetAwaiter().GetResult();
+                Resolved = true;
+            }
+
+            return _addresses.Length > 0;
+        }
+
+        /// <summary>
+        /// Task that looks up a Host name and returns its IP addresses.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        private async Task ResolveHostInternal()
+        {
+            if (!string.IsNullOrEmpty(HostName))
```

Comment:
Shouldn't you lock here? Or does it not matter if multiple threads try to resolve at once?

Reply:
Doesn't matter.  Thread safe.